.github/scripts/set-pr-status.js

```diff
@@ -0,0 +1,261 @@
+            updateProjectV2ItemFieldValue(input: {
+            projectId: "${projectId}",
+            itemId: "${itemId}",
+            fieldId: "${field.id}",
+            value: { date: "${date}" }
+            }) {
+            projectV2Item { id }
+            }
+        }
+        `);
+
+        console.log(`Set date field '${field.name}' to '${date}'.`);
+    }
+
+    // Sets the field for status to a project item (PR or Issue). E.g., Set Status for PR #100 to "In Progress".
+    async function setProjectItemStatus(itemId, field, optionName, projectId) {
+        // Ensure the field is a single select and has options
+        if (!field.options) throw new Error(`Field '${field.name}' has no options`);
+
+        // Find the matching option
+        const option = field.options.find(opt => opt.name === optionName);
+        if (!option) throw new Error(`Option '${optionName}' not found in field '${field.name}'`);
+
+        // Run the mutation
+        await github.graphql(`
+        mutation($projectId: ID!, $itemId: ID!, $fieldId: ID!, $optionId: String!) {
+            updateProjectV2ItemFieldValue(input: {
+            projectId: $projectId,
+            itemId: $itemId,
+            fieldId: $fieldId,
+            value: { singleSelectOptionId: $optionId }
+            }) {
+            projectV2Item { id }
+            }
+        }
+        `, {
+            projectId,
+            itemId,
+            fieldId: field.id,
+            optionId: option.id
+        });
+    }
+
+    // Gets all issue numbers that would be closed if the PR is merged.
+    function extractLinkedIssueNumbers(prBody) {
+        const body = prBody || "";
+        const withoutComments = body.replace(/<!--[\s\S]*?-->/g, "");
+        const matches = [...withoutComments.matchAll(/(?:close[sd]?|fixe[sd]?|resolve[sd]?)\s+#(\d+)/gi)];
+        return matches.map(match => parseInt(match[1], 10));
+    }
+
+    function findItemByNumber(allItems, number) {
+        return allItems.find(i => i.content?.number === number);
+    }
+
+    function getFieldByName(fields, fieldName) {
+        const field = fields.find(f => f.name === fieldName);
+        if (!field) throw new Error(`Field '${fieldName}' not found`);
+        return field;
+    }
+
+    // ========== MAIN LOGIC ==========
+    const pr = context.payload.pull_request;
+
+    // Run the status check only if the PR contains the l1 label.
+    // This check is for the submitted review, the check for a pull_request event was previously done.
+    if (context.eventName === "pull_request_review") {
+        const prNumber = pr.number;
+        const { data: labels } = await github.rest.issues.listLabelsOnIssue({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            issue_number: prNumber,
+        });
+
+        const hasL1 = labels.some(label => label.name === "L1");
+        if (!hasL1) {
+            console.log("PR does not have 'L1' label. Exiting.");
```

Comment:
since this action is l1 specifiy, maybe it should be stated in the workflow name, like `pr_github_status_l1`